internal/jbig2/segments/symbol-dictionary.go

```diff
@@ -0,0 +1,1127 @@
+	if s.isHuffmanEncoded {
+		return s.decodeHeightClassDeltaHeightWithHuffman()
+	}
+
+	i, err := s.arithmeticDecoder.DecodeInt(s.cxIADH)
+	if err != nil {
+		return 0, err
+	}
+
+	return int64(i), nil
+}
+
+// decodeHeightClassDeltaHeightWithHuffman - 6.5.6 decodes the symbol dictionary
+// when the height class is encoded using huffman tables.
+func (s *SymbolDictionary) decodeHeightClassDeltaHeightWithHuffman() (int64, error) {
+	switch s.sdHuffDecodeHeightSelection {
+	case 0:
+		t, err := huffman.GetStandardTable(4)
+		if err != nil {
+			return 0, err
+		}
+		return t.Decode(s.r)
+	case 1:
+		t, err := huffman.GetStandardTable(5)
+		if err != nil {
+			return 0, err
+		}
+		return t.Decode(s.r)
+	case 3:
+		if s.dhTable == nil {
+			t, err := huffman.GetStandardTable(0)
+			if err != nil {
+				return 0, err
+			}
+			s.dhTable = t
+		}
+		return s.dhTable.Decode(s.r)
+	}
+	return 0, nil
+}
+
+func (s *SymbolDictionary) decodeHeightClassCollectiveBitmap(
+	bmSize int64, heightClassHeight, totalWidth int,
+) (*bitmap.Bitmap, error) {
+	if bmSize == 0 {
+		heightClassColleciveBitmap := bitmap.New(totalWidth, heightClassHeight)
+		var (
+			b   byte
+			err error
+		)
+		for i := 0; i < len(heightClassColleciveBitmap.Data); i++ {
+			b, err = s.r.ReadByte()
+			if err != nil {
+				return nil, err
+			}
+
+			if err = heightClassColleciveBitmap.SetByte(i, b); err != nil {
+				return nil, err
+			}
+		}
+		return heightClassColleciveBitmap, nil
+	}
+	if s.genericRegion == nil {
+		s.genericRegion = NewGenericRegion(s.r)
+	}
+
+	s.genericRegion.setParameters(true, s.r.StreamPosition(), bmSize, heightClassHeight, totalWidth)
+
+	bm, err := s.genericRegion.GetRegionBitmap()
+	if err != nil {
+		return nil, err
+	}
+	return bm, nil
+}
+
+// getSbSymCodeLen 6.5.8.2.3 sets the SBSYMCODES variable.
+func (s *SymbolDictionary) getSbSymCodeLen() int {
+	first := int(math.Ceil(
+		math.Log(float64(s.numberOfImportedSymbols+s.numberOfNewSymbols)) / math.Log(2)))
+
+	if s.isHuffmanEncoded && first < 1 {
+		return 1
+	}
+
+	return first
+}
+
+func (s *SymbolDictionary) getToExportFlags() ([]int, error) {
+	var (
+		currentExportFlag int
+		exRunLength       int
+		err               error
+		totalNewSymbols   = s.numberOfImportedSymbols + s.numberOfNewSymbols
+		exportFlags       = make([]int, totalNewSymbols)
+	)
+
+	for exportIndex := 0; exportIndex < totalNewSymbols; exportIndex += exRunLength {
+		if s.isHuffmanEncoded {
+			t, err := huffman.GetStandardTable(1)
+			if err != nil {
+				return nil, err
+			}
+
+			i, err := t.Decode(s.r)
+			if err != nil {
+				return nil, err
+			}
+			exRunLength = int(i)
+		} else {
+			exRunLength, err = s.arithmeticDecoder.DecodeInt(s.cxIAEX)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		if exRunLength != 0 {
+			for index := exportIndex; index < exportIndex+exRunLength; index++ {
+				exportFlags[index] = currentExportFlag
+			}
+		}
+
+		if currentExportFlag == 0 {
```

Comment:
Could assign `0` to `currentExportFlag` before the if statement and remove the else branch.